packages/webapp/src/containers/Finances/RevenueDetail/index.jsx

```diff
@@ -0,0 +1,143 @@
+/*
+ *  Copyright 2023 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import React, { useState, useEffect } from 'react';
+import { deleteSale, updateSale } from '../actions';
+import { revenueByIdSelector } from '../selectors';
+import { revenueTypeByIdSelector } from '../../revenueTypeSlice';
+import { useDispatch, useSelector } from 'react-redux';
+import { useTranslation } from 'react-i18next';
+import { useCurrencySymbol } from '../../hooks/useCurrencySymbol';
+import { setPersistedPaths } from '../../hooks/useHookFormPersist/hookFormPersistSlice';
+import { HookFormPersistProvider } from '../../hooks/useHookFormPersist/HookFormPersistProvider';
+import useSortedRevenueTypes from '../AddSale/RevenueTypes/useSortedRevenueTypes';
+import GeneralRevenue from '../../../components/Forms/GeneralRevenue';
+import useCropSaleInputs, { getCustomFormChildrenDefaultValues } from '../useCropSaleInputs';
+import { hookFormPersistSelector } from '../../hooks/useHookFormPersist/hookFormPersistSlice';
+import useHookFormPersist from '../../hooks/useHookFormPersist';
+import { revenueTypeTileContentsSelector } from '../../revenueTypeSlice';
+
+function SaleDetail({ history, match }) {
+  const isEditing = match.path.endsWith('/edit');
+  const { sale_id } = match.params;
+
+  // To clear form history after editing
+  useHookFormPersist();
+  const { t } = useTranslation(['translation', 'revenue']);
+  const dispatch = useDispatch();
+
+  const revenueTypes = useSelector(revenueTypeTileContentsSelector);
+  // Doesnt populate unless used with hook form persist (but consider adding back persist because while this form doesnt need to persist -- others might want to)
+  // const revenueTypes = useSortedRevenueTypes();
```

Comment:
Let me try again -- I wanted to use this one but yes I was seeing that it needed an extra render cycle or something to populate the react select. Possible I just messed up somewhere. 

When you added it to GeneralRevenue was the main benefit of the useSortedRevenueTypes hook that it calls getRevenueTypes() ? Just in case the person did not follow the Finances store data loading path? The other stuff looks like a normal selector/data filter right?

Reply:
So yes it breaks, it does not populate the types and takes an extra render cycle ("other" in the console photo).
![Screenshot 2023-10-19 at 4 41 00 PM](https://github.com/LiteFarmOrg/LiteFarm/assets/30075973/80f58ef1-ec21-4cca-9229-2ba024cefdd2)
![Screenshot 2023-10-19 at 4 43 49 PM](https://github.com/LiteFarmOrg/LiteFarm/assets/30075973/ef7dcc58-2bf7-4788-b3de-48e608f6f497)

I think selectors are maybe better for sorting but the populating types store could be useful in cases where the data path is not followed.

Reply:
> When you added it to GeneralRevenue was the main benefit of the useSortedRevenueTypes hook that it calls getRevenueTypes() ? Just in case the person did not follow the Finances store data loading path? The other stuff looks like a normal selector/data filter right?

I wrote it so that it works no matter where it is called from. So the reason `getRevenueTypes()` is there is _Just in case the person did not follow the Finances store data loading path_ as you mentioned.

Do you remember we had that problem that a selector was crashing the app (we spent a good amount of time looking into it during tech daily, the selector was expecting tasks, crop varieties etc. I can't remember exactly), so I think we want to avoid adding an extra functionality to selectors.

The problem was that `revenue_type_option` is not calculated fast enough to make it to the initialization of the form. So what should be done is to `setValue(REVENUE_TYPE_OPTION, <selectedType>)`  when `revenueTypeOptions` is ready.

It worked by adding these to `GeneralRevenue`:
```
useEffect(() => {
    if (revenueTypeOptions?.length && !selectedTypeOption) {
      setValue(
        REVENUE_TYPE_OPTION,
        revenueTypeOptions?.find((option) => {
          return option.value === data[REVENUE_TYPE_ID];
        }),
      );
    }
  }, [revenueTypeOptions]);
```

I think using the new selector is better for now.

Reply:
Thanks!

Reply:
It seems to be working fine...!
I think now you can remove the imports of `revenueTypeTileContentsSelector`, and maybe the selector itself can also be removed?

Sorry if you would like to, `?` in `revenueTypeOptions?.find(...` could also be removed.

Reply:
Removed the selector!